
# HG changeset patch
# User Matt Johnston <matt@ucc.asn.au>
# Date 1366121792 -28800
# Node ID f110d321fe7abb8864a1ab3268984f7cb02ac285
# Parent  029e769c33c697f9255c40d6552f0aa81466eeed
Fix build when zlib is disabled, from
http://freetz.org/browser/trunk/make/dropbear/patches/350-no_zlib_fix.patch

diff -r 029e769c33c6 -r f110d321fe7a common-kex.c
--- a/common-kex.c	Mon Apr 15 22:11:11 2013 +0800
+++ b/common-kex.c	Tue Apr 16 22:16:32 2013 +0800
@@ -171,14 +171,18 @@
 	}
 	if (ses.kexstate.recvnewkeys && ses.newkeys->recv.valid) {
 		TRACE(("switch_keys recv"))
+#ifndef DISABLE_ZLIB
 		gen_new_zstream_recv();
+#endif
 		ses.keys->recv = ses.newkeys->recv;
 		m_burn(&ses.newkeys->recv, sizeof(ses.newkeys->recv));
 		ses.newkeys->recv.valid = 0;
 	}
 	if (ses.kexstate.sentnewkeys && ses.newkeys->trans.valid) {
 		TRACE(("switch_keys trans"))
+#ifndef DISABLE_ZLIB
 		gen_new_zstream_trans();
+#endif
 		ses.keys->trans = ses.newkeys->trans;
 		m_burn(&ses.newkeys->trans, sizeof(ses.newkeys->trans));
 		ses.newkeys->trans.valid = 0;

